src/prival.c

```diff
@@ -58,6 +58,50 @@ stumpless_get_prival_string( int prival ) {
   return prival_string;
 }
 
+const char *
+stumpless_get_priority_string( int prival ) {
+  const char *facility;
+  const char *severity;
+
+  size_t priority_string_size;
+  char* priority_string;
+
+  if ((prival & 0xff) != prival) {
```

Comment:
You'll need to use the `facility_is_invalid` and `severity_is_invalid` functions to perform this check, since there are some values that are not valid privals despite fitting into a single byte.

Reply:
Right! I've just amended the last commit and push forced the changes.

Reply:
Just a second... The tests are failing

Reply:
If I change the verification to:
`if ((facility_is_invalid(get_facility(prival))) || (severity_is_invalid(get_severity(prival)))) {`

then the negative integer test argument will always fail, since both getters mask the input.
Should I remove this test and keep the 0x100 one?

Reply:
Apparently both -1 and 0x100 doesn't work in this case.

Reply:
I would combine the test that you have with the invalid facility one. Thinking further, if you are testing for the lower byte mask, then the severity test will always pass since all 8 values are possible. The facility is the only one with additional things to worry about.

Also, it may be cleaner to factor this out so that the other prival functions can use it in the future.

Reply:
It looks like `stunpless_set_entry_prival` suffers from the same bug you just pointed out. I will open a separate issue for that when I am at a terminal again.

Reply:
I've just implemented a brief solution to this.
Tests are passing, things should work now.
I'm checking of byte range and also for facility range (the multiple of 8 wasn't necessary to test, since all masked values that are generated from `get_facility` are multiple of 8).

Let me know if any more corrections are needed.